src/test/scala/uk/gov/nationalarchives/dp/client/EntityClientTest.scala

```diff
@@ -1267,12 +1272,143 @@ abstract class EntityClientTest[F[_], S](preservicaPort: Int, secretsManagerPort
     val client = testClient(s"http://localhost:$preservicaPort")
     val response = valueFromF(
       cme.attempt(
-        client.entitiesByIdentifier("testIdentifier", "testValue", secretName)
+        client.entitiesByIdentifier(Identifier("testIdentifier", "testValue"), secretName)
       )
     )
 
     response.left.map { err =>
       err.getClass.getSimpleName should equal("PreservicaClientException")
     }
   }
+
+  "addIdentifiersForEntity" should s"make a correct request to add an identifier to an Entity" in {
+    val entityResponse =
+      <IdentifiersResponse xmlns="http://preservica.com/EntityAPI/v6.5" xmlns:xip="http://preservica.com/XIP/v6.5">
+        <Identifiers>
+          <xip:Identifier>
+            <xip:ApiId>65862d40f40440de14c1b75e5f342e99</xip:ApiId>
+            <xip:Type>TestIdentifierName</xip:Type>
+            <xip:Value>TestIdentifierValue</xip:Value>
+            <xip:Entity>{ref}</xip:Entity>
+          </xip:Identifier>
+        </Identifiers>
+        <Paging>
+          <TotalResults>1</TotalResults>
+        </Paging>
+        <AdditionalInformation>
+          <Self>http://mockapi.com/api/entity/structural-objects/$ref/identifiers</Self>
+        </AdditionalInformation>
+      </IdentifiersResponse>.toString()
+
+    preservicaServer.stubFor(post(urlEqualTo(tokenUrl)).willReturn(ok(tokenResponse)))
+    preservicaServer.stubFor(
+      post(urlEqualTo(s"/api/entity/structural-objects/$ref/identifiers")).willReturn(ok(entityResponse))
+    )
+
+    val client = testClient(s"http://localhost:$preservicaPort")
+    val addIdentifiersForEntityResponse: F[String] =
+      client.addIdentifiersForEntity(ref, "structural-objects", List(identifier), secretName)
+
+    val _ = valueFromF(addIdentifiersForEntityResponse)
+
+    val requestMade = preservicaServer.getServeEvents.getServeEvents.get(0).getRequest.getBody.map(_.toChar).mkString
```

Comment:
There's a `getBodyAsString` method so you don't have to convert from byte array.